src/main/java/de/tum/in/www1/artemis/web/rest/dto/ExerciseDetailStudentParticipationDTO.java

```diff
@@ -0,0 +1,97 @@
+package de.tum.in.www1.artemis.web.rest.dto;
+
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.artemis.domain.ProgrammingSubmission;
+import de.tum.in.www1.artemis.domain.Result;
+import de.tum.in.www1.artemis.domain.Submission;
+import de.tum.in.www1.artemis.domain.Team;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.enumeration.AssessmentType;
+import de.tum.in.www1.artemis.domain.enumeration.InitializationState;
+import de.tum.in.www1.artemis.domain.enumeration.SubmissionType;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseStudentParticipation;
+import de.tum.in.www1.artemis.domain.participation.StudentParticipation;
+
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public record ExerciseDetailStudentParticipationDTO(long id, InitializationState initializationState, ZonedDateTime initializationDate, ZonedDateTime individualDueDate,
+        boolean testRun, String type, Set<ResultDTO> results, Set<SubmissionDTO> submissions, UserDTO student, TeamDTO team,
```

Comment:
Why are results and submissions included here when submissions can also contain results as well?
I know that this is the case for the entities, but what is the reason to do the same here and potentially send the same data twice?